public/app/features/apiserver/client.ts

```diff
@@ -38,18 +38,26 @@ export class ScopedResourceClient<T = object, S = object, K = string> implements
     return getBackendSrv().get<Resource<T, S, K>>(`${this.url}/${name}`);
   }
 
-  public watch(opts?: WatchOptions): Observable<ResourceEvent<T, S, K>> {
+  public watch(
+    params?: WatchOptions,
+    config?: Pick<BackendSrvRequest, 'data' | 'method'>
+  ): Observable<ResourceEvent<T, S, K>> {
     const decoder = new TextDecoder();
-    const params = {
-      ...opts,
+    const { name, ...rest } = params ?? {}; // name needs to be added to fieldSelector
+    const requestParams = {
+      ...rest,
       watch: true,
-      labelSelector: this.parseListOptionsSelector(opts?.labelSelector),
-      fieldSelector: this.parseListOptionsSelector(opts?.fieldSelector),
+      labelSelector: this.parseListOptionsSelector(params?.labelSelector),
+      fieldSelector: this.parseListOptionsSelector(params?.fieldSelector),
     };
+    if (name) {
+      requestParams.fieldSelector = `metadata.name=${name}`;
+    }
     return getBackendSrv()
       .chunked({
-        url: params.name ? `${this.url}/${params.name}` : this.url,
-        params,
+        url: this.url,
```

Comment:
I think you need to build the URL based on name and path.

Reply:
I thought so... *BUT* it turns out that watch only works with the list command and you have to send name as: `fieldSelector=metadata.name={name}`